tensorboard/webapp/core/store/core_types.ts

```diff
@@ -32,6 +34,11 @@ export interface CoreState {
   // Size of a page in a general paginated view that is configurable by user via
   // settings.
   pageSize: number;
+
+  // TODO(stephanwlee): move these state to the `runs` features.
+  // For now, we want them here for Polymer interp states reasons, too.
+  polymerInteropRuns: Run[];
```

Comment:
Can you explain to me (1) why you don't immediately structure the data model the way you want (with `runs` as a feature) and (2) why these are prefixed with polymerInterop.

Reply:
Ah, sure. I think the question here is related to the one on the core_actions.ts, too. 

The notion of a run is native on the internal application, but it is not the case for OSS. Although unclear when, we will eventually port the features into OSS. While waiting for the feature to be ported out, I wanted to unblock some of our OSS Angular development by making a notion of runs first class in OSS too. However, since the name of the selector/action will be about the same, I did not want them to eventually collide and make troubles. Also, I wanted to make sure the readers are aware that this state is temporary and is to be only used for interop with the Polymer component, tf-runs-selector.

While the shape and API is nearly congruent as the internal one for now, I did not want to make porting harder in case they go out of sync.

Now regarding #1, there is a cost associated with creating a new feature:
1. code bloat is one but...
2. unclear how it will reconcile with internal `runs` feature when the names are the same. 

Does this answer both of your questions?

Reply:
Yes, I think so.  Still wondering a bit about the questions in my other comment but I think this offers enough clues.